src/main/java/seedu/address/logic/commands/SearchPatientCommand.java

```diff
@@ -0,0 +1,197 @@
+                    isPersonFound = true;
+
+                }
+            }
+            if (!isPersonFound) {
+                throw new CommandException(MESSAGE_PERSON_NOT_FOUND);
+            } else {
+                return new CommandResult(String.format(MESSAGE_SEARCH_PERSON_SUCCESS, personFound));
+            }
+        }
+
+        else if (areaToSearch == areaIsTemperature) {
+            double startingTemperature = searchPatientDescriptor.getTemperatureRange().getStartingTemperature();
+            double endingTemperature = searchPatientDescriptor.getTemperatureRange().getEndingTemperature();
+            for (Person person : personList) {
+                if (person.getTemperature().getValue() >= startingTemperature
+                        && person.getTemperature().getValue() <= endingTemperature) {
+                    personWithinTemperatureRange.add(person);
+                }
+            }
+
+            if (personWithinTemperatureRange.isEmpty()) {
+                throw new CommandException(MESSAGE_PERSON_NOT_FOUND);
+            } else {
+                return new CommandResult(MESSAGE_SEARCH_PERSON_LIST_SUCCESS
+                        + getListOutput(personWithinTemperatureRange));
+            }
+        } else if (areaToSearch == tooManyArea) {
+            return new CommandResult(Messages.MESSAGE_TOO_MANY_COMMANDS);
+        }
+        return null;
+    }
+
+    /**
+     * Return the list of persons' details
+     * @param list a list that stores the filtered persons.
+     * @return a String output of the persons' details.
+     */
+    public String getListOutput(ArrayList<Person> list) {
+        StringBuilder outputString = new StringBuilder();
+        for (int i = 0; i < list.size(); i++) {
+            outputString.append(String.format("%d.%s\n", i + 1, list.get(i)));
+        }
+        return outputString.toString();
+    }
+
+    /**
+     * Confirm the area to look for.
+     * @param searchPatientDescriptor Details of the searchPatient Command.
+     * @return area to look for.
+     */
+    public int confirmArea(SearchPatientDescriptor searchPatientDescriptor) {
+        if (searchPatientDescriptor.getOptionalName().isEmpty()
+                && searchPatientDescriptor.getOptionalTemperatureRange().isEmpty()) {
+            return areaNotFound;
+        }
+        else if (searchPatientDescriptor.getOptionalName().isPresent()
+                && searchPatientDescriptor.getOptionalTemperatureRange().isEmpty()) {
+            return areaIsName;
+        }
+        else if (searchPatientDescriptor.getOptionalName().isEmpty()
+                && searchPatientDescriptor.getOptionalTemperatureRange().isPresent()) {
+            return areaIsTemperature;
+        }
+        else {
+            return tooManyArea;
+        }
+
+    }
+
+    /**
+     * Stores the details to search the person .
+     * The class is used with @ConfirmArea method to confirm the area to look for.
```

Comment:
JavaDoc comments here are really useful to understand the purpose of the nested class. Great work 👍